engine/src/main/java/org/pentaho/di/pan/Pan.java

```diff
@@ -239,48 +244,86 @@ public static void main( String[] a ) throws Exception {
-              .namedParams( optionParams )
-              .build();
-
-      Result result = getCommandExecutor().execute( transParams, args.toArray( new String[ args.size() ] ) );
-
-      exitJVM( result.getExitStatus() );
+        .blockRepoConns( optionNorep.toString() )
+        .repoName( optionRepname.toString() )
+        .repoUsername( optionUsername.toString() )
+        .trustRepoUser( optionTrustUser.toString() )
+        .repoPassword( optionPassword.toString() )
+        .inputDir( optionDirname.toString() )
+        .inputFile( optionTransname.toString() )
+        .listRepoFiles( optionListtrans.toString() )
+        .listRepoDirs( optionListdir.toString() )
+        .exportRepo( optionExprep.toString() )
+        .localFile( optionFilename.toString() )
+        .localJarFile( optionJarFilename.toString() )
+        .localInitialDir( initialDir.toString() )
+        .listRepos( optionListrep.toString() )
+        .safeMode( optionSafemode.toString() )
+        .metrics( optionMetrics.toString() )
+        .listFileParams( optionListParam.toString() )
+        .logLevel( "" )
+        .maxLogLines( "" )
+        .maxLogTimeout( "" )
+        .logFile( "" )
+        .oldLogFile( "" )
+        .version( "" )
+        .resultSetStepName( optionResultSetStepName.toString() )
+        .resultSetCopyNumber( optionResultSetCopyNumber.toString() )
+        .base64Zip( optionBase64Zip.toString() )
+        .namedParams( optionParams )
+        .build();
+
+      Result rslt = getCommandExecutor().execute( transParams, args.toArray( new String[ args.size() ] ) );
+
+      exitJVM( rslt.getExitStatus() );
 
     } catch ( Throwable t ) {
       t.printStackTrace();
       exitJVM( CommandExecutorCodes.Pan.UNEXPECTED_ERROR.getCode() );
     }
   }
 
+  private static CommandExecutorResult validateAndSetPluginParam( LogChannelInterface log, String path ) {
+//    String returnMsg = "";
+    CommandExecutorResult result = null;
+      try {
+        Collection<CommandLineOptionProvider> commandlineProviders = PluginServiceLoader.loadServices( CommandLineOptionProvider.class );
+        CommandLineOptionProvider provider = commandlineProviders.stream().findFirst().orElse( null );
+        result = provider.handleParameter( log, path );
+
+      } catch ( KettlePluginException ke ) {
+      } catch ( KettleException e ) {
+        throw new RuntimeException( e );
+      }
+      return result;
+  }
+
+  private static List<CommandLineOption> getAdditionalCommandlineOption( CommandLineOption[] options,
+                                                                         StringBuilder param ) {
+    List<CommandLineOption> modifiableList = new ArrayList<>();
+
+    Collections.addAll( modifiableList, options );
+    try {
+      for ( CommandLineOptionProvider provider : PluginServiceLoader.loadServices( CommandLineOptionProvider.class ) ) {
+        provider.prepareAdditionalCommandlineOption( modifiableList, param );
+      }
```

Comment:
it's still not clear to me how this can re-use the param multiple times. What is param doing?

Reply:
currently it's not handling multiple command line option as it's not a specified requirement in the story. 
Maybe we can add later?!

Reply:
currently it's not handling multiple command line option as it's not a specified requirement in the story. 
Maybe we can add later?!

Reply:
add a comment, then, so a future developer won't be confused. :)